multiversx_sdk/transfers/transfer_transactions_factory.py

```diff
@@ -104,10 +114,7 @@ def create_transaction_for_transfer(
         token_transfers: Optional[list[TokenTransfer]] = None,
         data: Optional[bytes] = None,
     ) -> Transaction:
-        if token_transfers and data:
-            raise BadUsageError("Can't set data field when sending esdt tokens")
-
-        if (native_amount and not token_transfers) or data:
+        if (native_amount or data) and not token_transfers:
```

Comment:
Not sure, can we make the checks a bit easier to understand by starting with `if token transfers do that, else do that`? Or maybe it complicates things?

Reply:
done